### PR TITLE
Set wiki link per user's selected language

### LINKED ISSUE #815: Update Wiki Footer Link to use User's Language
Currently, when a user clicks on the Scratch Wiki footer link, it takes them to the English wiki. This ignores the user's language. It would be nice if when a user clicks on the Scratch Wiki footer link, they are taken to the wiki of their language, if it exists.

### PR DESCRIPTION
**Pull request by R4356th**
### Resolves:

Resolves #815.

### Changes:

Selects Scratch Wiki link according to the user's selected language.

### CODE REVIEW

**Comment by R4356th** [comment]
@benjiwheeler (I forgot to that in the first place, sorry.)

**Comment by R4356th** [comment]
@apple502j, the pr is likely not be merged soon anyway. I am going to mark it as ready for review, which is true state.

**Comment by R4356th** [comment]
@apple502j, the pr is likely not be merged soon anyway. I am going to mark it as ready for review, which is true state.

**Comment by R4356th** [comment]
@Kenny2github, alright.

**Comment by R4356th** [comment]
Done, @Kenny2github.

File: src/components/footer/www/footer.jsx [hunk 1]
```diff
@@ -11,6 +11,22 @@ const frameless = require('../../../lib/frameless');
...
+            'hu': 'https://hu.scratch-wiki.info/',
+            'fr': 'https://fr.scratch-wiki.info/'
+        };
+        var wiki_link = null;
+        if (this.props.intl.locale in interwiki_mapping) {
+            wikilink = interwiki_mapping[this.props.intl.locale];
+        } else {
+            wikilink = 'https://en.scratch-wiki.info';
+        }
+
```
Comment (Kenny2github) [inline src/components/footer/www/footer.jsx:29]:
```suggestion
const interwikiMapping = {
    de: 'https://de.scratch-wiki.info',
    ru: 'https://ru.scratch-wiki.info/',
    nl: 'https://nl.scratch-wiki.info/',
    id: 'https://id.scratch-wiki.info/',
    ja: 'https://ja.scratch-wiki.info/',
    hu: 'https://hu.scratch-wiki.info/',
    fr: 'https://fr.scratch-wiki.info/'
};
var wikiLink = null;
if (this.props.intl.locale in interwiki_mapping) {
    wikiLink = interwikiMapping[this.props.intl.locale];
} else {
    wikiLink = 'https://en.scratch-wiki.info';
}
```

File: src/components/footer/www/footer.jsx [hunk 2]
```diff
@@ -108,7 +124,7 @@ const Footer = props => (
                         </a>
                     </dd>
                     <dd>
-                        <a href="https://en.scratch-wiki.info/">
+                        <a href="{wikilink}">
```
Comment (Kenny2github) [inline src/components/footer/www/footer.jsx:127]:
```suggestion
                        <a href="{wikiLink}">
```
(to match with other suggested code)

File: src/components/footer/www/footer.jsx [hunk 3]
```diff
@@ -11,6 +11,22 @@ const frameless = require('../../../lib/frameless');
...
+            'ru': 'https://ru.scratch-wiki.info/',
+            'nl': 'https://nl.scratch-wiki.info/',
+            'id': 'https://id.scratch-wiki.info/',
+            'ja': 'https://ja.scratch-wiki.info/',
+            'hu': 'https://hu.scratch-wiki.info/',
+            'fr': 'https://fr.scratch-wiki.info/'
+        };
+        var wiki_link = null;
+        if (this.props.intl.locale in interwiki_mapping) {
+            wikilink = interwiki_mapping[this.props.intl.locale];
```
Comment (Kenny2github) [inline src/components/footer/www/footer.jsx:25]: Why is there a `this` here? There's no object to access.

Comment (R4356th) [reply src/components/footer/www/footer.jsx]: On line 24 or 25?

Comment (Kenny2github) [reply src/components/footer/www/footer.jsx]: Both - Travis is saying "Unexpected `this`", and for good reason - you're not in an object's method, so what could `this` possibly refer to?

File: src/components/footer/www/footer.jsx [hunk 4]
```diff
@@ -11,6 +11,21 @@ const frameless = require('../../../lib/frameless');
...
+    de: 'https://de.scratch-wiki.info',
+    ru: 'https://ru.scratch-wiki.info/',
+    nl: 'https://nl.scratch-wiki.info/',
+    id: 'https://id.scratch-wiki.info/',
+    ja: 'https://ja.scratch-wiki.info/',
+    hu: 'https://hu.scratch-wiki.info/',
+    fr: 'https://fr.scratch-wiki.info/'
+};
+var wikiLink = null;
+if (this.props.intl.locale in interwiki_mapping) {
```
Comment (Kenny2github) [inline src/components/footer/www/footer.jsx:24]: Forgot to change this varname
```suggestion
if (this.props.intl.locale in interwikiMapping) {
```

**Comment by Kenny2github** [comment]
I think I've reached the limit of how I can help here. @chrisgarrity or @benjiwheeler how does one access the user's locale at the top level? What should be done here?

File: src/components/footer/www/footer.jsx (hunk 3 above)
Comment (R4356th) [reply src/components/footer/www/footer.jsx]: Well, @Kenny2github, I was thinking that it was needed to get the user's language.

**Comment by apple502j** [comment]
@R4356th I also made a PR, #4121 - you can steal mine :P

File: src/components/footer/www/footer.jsx [hunk 5]
```diff
@@ -20,8 +20,8 @@ const interwikiMapping = {
     hu: 'https://hu.scratch-wiki.info/',
     fr: 'https://fr.scratch-wiki.info/'
 };
-var wikiLink = null;
-if (this.props.intl.locale in interwikiMapping) {
+const wikiLink = null;
+if (props.intl.locale in interwikiMapping) {
```
Comment (apple502j) [inline src/components/footer/www/footer.jsx:24]: props is only available inside `const Footer =`

Comment (R4356th) [reply src/components/footer/www/footer.jsx:24]: Oh, thanks.

**Comment by benjiwheeler** [comment]
Thanks @R4356th -- I'm closing this and planning to merge https://github.com/LLK/scratch-www/pull/4121 instead. Thanks for getting this to happen!

**Comment by R4356th** [comment]
Alright, @benjiwheeler. But a question though- aren't we supposed to make the least possible amount of changes?

**Pull request closed without merging**